pagerduty/resource_pagerduty_incident_workflow.go

```diff
@@ -322,41 +449,41 @@ func flattenIncidentWorkflow(d *schema.ResourceData, iw *pagerduty.IncidentWorkf
 	}
 
 	if includeSteps {
-		steps := flattenIncidentWorkflowSteps(iw, nonGeneratedInputNames)
+		steps := flattenIncidentWorkflowSteps(iw, specifiedSteps)
 		d.Set("step", steps)
 	}
 
 	return nil
 }
 
-func flattenIncidentWorkflowSteps(iw *pagerduty.IncidentWorkflow, nonGeneratedInputNames map[string][]string) []map[string]interface{} {
+func flattenIncidentWorkflowSteps(iw *pagerduty.IncidentWorkflow, specifiedSteps []*SpecifiedStep) []map[string]interface{} {
 	newSteps := make([]map[string]interface{}, len(iw.Steps))
 	for i, s := range iw.Steps {
-		nonGeneratedInputNamesForStep, ok := nonGeneratedInputNames[s.ID]
-		if !ok {
-			nonGeneratedInputNamesForStep = make([]string, 0)
-		}
-
 		m := make(map[string]interface{})
+		specifiedStep := *specifiedSteps[i]
 		m["id"] = s.ID
 		m["name"] = s.Name
 		m["action"] = s.Configuration.ActionID
-		m["input"] = flattenIncidentWorkflowStepInput(s.Configuration.Inputs, nonGeneratedInputNamesForStep)
+		m["input"] = flattenIncidentWorkflowStepInput(s.Configuration.Inputs, specifiedStep.SpecifiedInputNames)
+		m["inline_steps_input"] = flattenIncidentWorkflowStepInlineStepsInput(
+			s.Configuration.InlineStepsInputs,
+			specifiedStep.SpecifiedInlineInputs,
+		)
 
 		newSteps[i] = m
 	}
 	return newSteps
 }
 
-func flattenIncidentWorkflowStepInput(inputs []*pagerduty.IncidentWorkflowActionInput, nonGeneratedInputNames []string) *[]interface{} {
+func flattenIncidentWorkflowStepInput(inputs []*pagerduty.IncidentWorkflowActionInput, specifiedInputNames []string) *[]interface{} {
 	newInputs := make([]interface{}, len(inputs))
 
 	for i, v := range inputs {
 		m := make(map[string]interface{})
 		m["name"] = v.Name
 		m["value"] = v.Value
 
-		if !isInputInNonGeneratedInputNames(v, nonGeneratedInputNames) {
+		if !isInputInNonGeneratedInputNames(v, specifiedInputNames) {
```

Comment:
Once we get to the point where we set generated=true or not, the logic is the same. Now that we pass the specified inputs recursive struct along as we flatten the response, we can easily access what the current specified inputs are.